src/polytope/github/repository/Repository.py

```diff
@@ -0,0 +1,380 @@
+import json
+from dataclasses import asdict
+from typing import TYPE_CHECKING, Optional, Tuple
+
+import re
+import requests
+
+from polytope.github import Requester
+from polytope.github.RequestVerb import RequestVerb
+from polytope.github.Session import RequestsSession
+from polytope.github.Session import Session
+
+from .InternalCode import GithubRepositoryInternalCode as GHIC
+from .RepositoryConfig import GithubRepositoryConfig
+from .Response import GithubRepositoryResponse
+
+# only for mypy.
+if TYPE_CHECKING:
+    from polytope.github import Token
+
+# alphanumeric or hyphen, starts & ends with alphanumeric
+GITHUB_USERNAME_REGEX = r"^[a-zA-Z\d](?:[a-zA-Z\d]|-(?=[a-zA-Z\d])){0,37}[a-zA-Z\d]$"
+# alphanumeric, hyphen, underscore. starts & ends with alphanumeric.
+GITHUB_REPONAME_REGEX = r"^[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?$"
+
+
+class GithubRepository:
+    """
+    Controller of Github Repository.
+
+    @param token: Personal Access Token.
+    @param owner: Github Username of Repository Owner.
+    @param name: Name of the repository.
+    """
+
+    def __init__(
+        self, owner: str, name: str, token: "Token", session: Session = RequestsSession
```

Comment:
(comment) The type of `session` should be `Type[Session]` not `Session`, I think.